src/app/pages/applications/chart-releases/chart-releases.component.ts

```diff
@@ -330,12 +330,29 @@ export class ChartReleasesComponent implements OnInit, OnDestroy {
   }
 
   update(name: string): void {
+    const catalogApp = this.chartItems[name];
     this.appLoaderService.open();
     this.appService.getUpgradeSummary(name).pipe(untilDestroyed(this)).subscribe((res: UpgradeSummary) => {
       this.appLoaderService.close();
+      let msg = `<div><strong>${helptext.charts.upgrade_dialog.currentVersion}:</strong> ${catalogApp.human_version}`;
+      msg += `<div><strong>${helptext.charts.upgrade_dialog.versionToUpgrade}:</strong> ${res.latest_human_version}`;
+      msg += `<div><strong>${helptext.charts.upgrade_dialog.images}:</strong></div>`;
+      if (Object.keys(res.container_images_to_update).length > 0) {
+        for (const key in res.container_images_to_update) {
+          msg += `<div>${key}</div>`;
+        }
+      } else {
+        msg += `${helptext.charts.upgrade_dialog.noImageUpdate}`;
+      }
+      msg += `<div><strong>${helptext.charts.upgrade_dialog.changeLog}</strong></div>`;
+      if (res.changelog) {
+        msg += res.changelog;
+      } else {
+        msg += helptext.charts.upgrade_dialog.noChangeLogMsg;
+      }
```

Comment:
I'm on the fence about joining strings like this in the component code.
I think it may be better to just add another component for this.
You can create something like UpdateDialogComponent, open it with `this.matDialog.open`, receive `updateSummary` in `@Inject(MAT_DIALOG_DATA)` and return values in `dialogRef.close(true/false)`.

Then you can construct the message cleanly in an html file.

For examples, see `SelectPresetModalComponent` and how it's being called in `DatasetAclEditorComponent.onUsePresetPressed`